File: presenter/src/commonMain/kotlin/ro/kofe/presenter/state/StateLoggerImpl.kt
package ro.kofe.presenter.state

import ro.kofe.model.Event
import ro.kofe.presenter.provider.LoggingProvider

class StateLoggerImpl(
    private val stateRepository: StateProvider,
    private val loggingProvider: LoggingProvider
) : StateLogger {

    override fun logState(unixTime: Long, event: Event): Boolean {
        TODO("Not yet implemented")
    }

    override fun getStateMap(): Map<Long, Event> {
        TODO("Not yet implemented")
    }
}

File: ios2/frames/provider/StateLoggerImpl.swift
//
//  StateLoggerImpl.swift
//  kofero
//
//  Created by Mitchell Drew on 24/1/24.
//

import Foundation
import presenter


//TODO: LOW PRIORITY
class StateLoggerImpl: StateLogger {
//    private let diskAccessor: DiskAccessor
//    private let mapper: StringMapper<[KotlinLong:ModelEvent]>
//
//    init(diskAccessor: DiskAccessor, mapper: StringMapper<[KotlinLong:ModelEvent]>){
//        self.diskAccessor = diskAccessor
//        self.mapper = mapper
//    }
    
    func getStateMap() -> [KotlinLong : ModelEvent] {
        return [KotlinLong:ModelEvent]()
    }
    
    func logState(unixTime: Int64, event: ModelEvent) {
        var map = getStateMap()
        map[KotlinLong(value: unixTime)] = event
        //diskAccessor.write(fileName: "statemap", json: )
    }
    
    
}
